pkg/agent/informant.go

```diff
@@ -1203,7 +1203,23 @@ func (s *InformantServer) MonitorHealthCheck(ctx context.Context, logger *zap.Lo
 	}
 
 	_, err = s.awaitMonitorResponse(ctx, rx)
-	return err
+	if err != nil {
+		return err
+	}
+
+	s.runner.lock.Lock()
+	defer s.runner.lock.Unlock()
+
+	// Update our record of the last successful time we heard from the monitor, if the "server"
```

Comment:
I would drop the "if the server is confirmed part" as connected to monitor implies server is confirmed, and we don't even check for this anyways.

Reply:
Makes sense - done! 4f04f81